src/inputs/netprobe/PingProbe.cpp

```diff
@@ -93,18 +93,29 @@ void PingReceiver::_setup_receiver()
                 timeval time;
                 TIMESPEC_TO_TIMEVAL(&time, &stamp);
                 pcpp::RawPacket raw(reinterpret_cast<uint8_t *>(_array.data()), rc, time, false, pcpp::LINKTYPE_DLT_RAW1);
-                pcpp::Packet packet(&raw, pcpp::ICMP);
-                std::lock_guard<std::recursive_mutex> lock(mutex);
-                recv_signal(packet, stamp);
+                _recv_packets.push_back({pcpp::Packet(&raw, pcpp::ICMP), stamp});
```

Comment:
do you look at the Packet raw data ownership model here? I recall there being various options to have the raw data freed or not when this Packet object is destroyed - just want to make sure we don't either double free or leak

Reply:
yes.. I did try other property and got double freed issue

Reply:
I've also ran the valgrind to check for memory leak and got none